ci(all): Enforce using Conventional Commit PR title

miquelbeltran opened the pull request:
## Description

Enforces the use of Conventional Commits on all PRs.

This ensures that all work can be managed using `melos version`, as it requires following Conventional Commits strictly.

Also makes reading PR changes easier.

I used the example provided by https://github.com/marketplace/actions/semantic-pull-request to configure it.

## Related Issues

None

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

vbuberen commented:
I think we need to move this check into a separate workflow, because `all_plugins` runs not for all commits (see `ignored-paths`), while we would want to be sure in all incoming commits.

Also, I see that the referenced workflow uses https://github.com/marketplace/actions/semantic-pull-request action and I would just use it directly.

miquelbeltran commented on .github/workflows/pr.yml:
```diff
@@ -0,0 +1,42 @@
+name: "Pull-Request Check"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
```
This workflow will run on PR changes, not code changes

miquelbeltran commented on .github/workflows/pr.yml:
```diff
@@ -0,0 +1,42 @@
+    steps:
+      - uses: amannn/action-semantic-pull-request@v4
+        id: lint_pr_title
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
I am not sure if this is required, I copied it over from the example

miquelbeltran commented:
@vbuberen I have addressed your points. I am unable to check if this really works as it doesn't seem to trigger if it is in a branch and not on main, so we would have to wait until merged to see it in action.

mhadaily commented:
This is great

nohli reviewed (approved):
Great idea, let's see if it works 😊
